sopel/tools/memories.py

```diff
@@ -166,3 +205,46 @@ def __contains__(self, key):
 
     def __setitem__(self, key: Optional[str], value):
         super().__setitem__(self._make_key(key), value)
+
+    def setdefault(self, key: str, default=None):
+        return super().setdefault(self._make_key(key), default)
+
+    def __delitem__(self, key: str):
+        super().__delitem__(self._make_key(key))
+
+    def copy(self):
+        return self.__class__(
+            self,
+            identifier_factory=self.make_identifier,
+        )
+
+    def get(self, key: str, default=_NO_DEFAULT):
+        if default is _NO_DEFAULT:
+            return super().get(self._make_key(key))
+        return super().get(self._make_key(key), default)
+
+    def pop(self, key: str, default=_NO_DEFAULT):
+        if default is _NO_DEFAULT:
+            return super().pop(self._make_key(key))
+        return super().pop(self._make_key(key), default)
+
+    def update(self, maybe_mapping=tuple()):
+        super().update(self._process_arg(maybe_mapping))
+
+    def __or__(self, other):
+        if not isinstance(other, dict):
+            return NotImplemented
+        new = type(self)(self, identifier_factory=self.make_identifier)
+        new.update(other)
+        return new
+
+    def __ror__(self, other):
+        if not isinstance(other, dict):
+            return NotImplemented
+        new = type(self)(other, identifier_factory=self.make_identifier)
```

Comment:
See previous remark

```suggestion
        new = self.copy()
```

Reply:
Oops, yea, that should have been `new = other.copy()` in my suggestion. Copy/paste error of my own…

Reply:
Why not just: 

```python
new = self.copy()
new.update({k: v for k, v in other.items() if k not in self)
```

Reply:
> Why not just:

I think it's less clear what that spelling does. Taking a copy of `other` and then updating it with `self` exactly mirrors how the `__or__` method works, and (to me) seems clearer even with the `type(self)(...)` shenanigans. Adding a comment as @SnoopJ suggested isn't a bad idea, especially since I think _either_ spelling would need one.

More importantly, the dict comprehension means iterating over `other` _twice_: once to build the argument to `update()` in an ephemeral `dict`, and then again when `update()` calls `_convert_keys()` on its input.

(One couldn't even bypass the second iteration by pre-converting the keys in the dict comprehension; `update()` will call `_convert_keys()` regardless of whether any keys in its input actually need conversion, since checking first would introduce _another_ iteration over all keys in the input.)

Reply:
Oh oh that's because I forgot that you can pass an iterable of tuple (k, v) to update and it'll work:

```python
new = self.copy()
new.update((k, v) for k, v in other.items() if k not in self)
```

Now it's a generator expression, so it's not evaluated twice: it's evaluated only once!

And I think it is clearer that way: get a copy of the current state, and add any key that isn't already there from the other. It mimics almost exactly `__or__` by adding a slight difference.

Reply:
Ah, after remembering generator expressions in one of my other commits this week, I forgot about them again when looking at this feedback. You, sir, earned a co-author credit. :)